subworkflows/nf-core/dia_proteomics_analysis/main.nf

```diff
@@ -0,0 +1,426 @@
+            def meta_enzyme_mods = meta_input.subMap(['enzyme', 'fixed_mods', 'variable_mods']) + [id: id_enzyme_mods]
+            def meta_fasta_enzyme = [id: meta_fasta.id + '_' + meta_enzyme_mods.id]
+            def meta_input_fasta = meta_input + meta_fasta + [id: meta_input.id + '_' + meta_fasta.id] + [experiment: meta_exp_searchdb]
+            meta_input.experiment = meta_exp_searchdb
+
+            expdesign: [meta_exp_searchdb, expdesign]
+
+            ms_file: [meta_input, meta_input_fasta, ms_file]
+            ms_file_fasta: [meta_fasta_enzyme, meta_input_fasta, ms_file]
+            search_db_by_exp: [meta_exp_searchdb, fasta]
+            search_db_by_enzyme: [meta_enzyme_mods, meta_fasta_enzyme, meta_fasta, fasta]
+            enzyme_mods: [meta_enzyme_mods, enzyme, fixed_mods, variable_mods]
+            mass_tolerance_settings: [meta_input_fasta, precursor_tolerance, fragment_tolerance, precursor_tolerance_unit, fragment_tolerance_unit, ms_file, fasta]
+
+            speclib: [meta_fasta_enzyme, speclib]
+            empirical: [meta_exp_searchdb, empirical_lib, empirical_log]
+        }
+
+    ch_speclib = input.speclib.filter{ tuple -> tuple[1] != null }
+    ch_empirical = input.empirical.filter{ tuple -> tuple[1] != null }
+
+    //
+    // MODULE: Generate DIA-NN configuration. Needs to run once each unique enzyme, fixed modifications, and variable modifications combination.
+    //
+    // Only run config generation and speclib generation if no speclib is provided
+    // Conditional execution pattern:
+    //   - combine with ifEmpty(null) pairs each input with either existing speclib or null
+    //   - filter keeps only tuples containing null (i.e., no pre-generated speclib exists)
+    //   - downstream processes only execute when needed
+
+    ch_config_input = input.enzyme_mods.unique() // [meta_enzyme_mods, enzyme, fixed_mods, variable_mods]
+        .combine(ch_speclib.ifEmpty(null))
+        .filter{tuple -> tuple.any { it == null }} 
+        .map{ meta, enzyme, fixed_mods, variable_mods, _ ->
+            [meta, enzyme, fixed_mods, variable_mods]
+        }
+    
+    QUANTMSUTILS_DIANNCFG(ch_config_input)
+    ch_versions = ch_versions.mix(QUANTMSUTILS_DIANNCFG.out.versions)
+
+    //
+    // MODULE: In-silico library generation. Needs to run once for every unique config/ FASTA dataset combination.
+    //
+
+    ch_insilico_library_input = input.search_db_by_enzyme    // [meta_enzyme_mods, meta_fasta_enzyme, meta_fasta, fasta]
+        .combine(QUANTMSUTILS_DIANNCFG.out.diann_cfg, by: 0) // [meta_enzyme_mods, meta_fasta_enzyme, meta_fasta, fasta, cfg_file]
+        .unique() // Multiple inputs might have the same config
+        .map { meta_enzyme_mods, meta_fasta_enzyme, meta_fasta, fasta, cfg_file -> 
+            [meta_fasta_enzyme + [config: cfg_file.text], [], [], fasta, [], []] // Added ms_file_names as []
```

Comment:
As we have one lib for each fasta, [enzyme], [fix_mods], [var_mods] combination, should we create a `meta_fasta_enzyme_mods` and use here?

Reply:
That's basically what this is doing with the `+`